### PR TITLE
Implement validator.js task

### PR DESCRIPTION
**Pull request by SergeiKalachev**
Implemented task: https://learn.javascript.ru/taskbook/nodejs-20220303/workspace/testing-configuration-logging/unit-tests

### CODE REVIEW

**Comment by jsru** [comment]
Добавляю преподавателя (@tadjik1) для код-ревью.

File: 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js [hunk 1]
```diff
@@ -3,20 +3,205 @@ const expect = require('chai').expect;
...
+      const errors = validator.validate({name: 1, age: ''});
+      expect(errors).to.have.length(2);
       expect(errors[0]).to.have.property('field').and.to.be.equal('name');
-      expect(errors[0]).to.have.property('error').and.to.be.equal('too short, expect 10, got 6');
+      expect(errors[0]).to.have.property('error').and.to.be.equal('expect string, got number');
+      expect(errors[1]).to.have.property('field').and.to.be.equal('age');
+      expect(errors[1]).to.have.property('error').and.to.be.equal('expect number, got string');
+    });
+
+    it('gives 0 errors for empty object', () => {
```
Comment (SergeiKalachev) [inline 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js:50]: Not sure if it is the desired implementation. Should we assume all fields are required?

File: 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js [hunk 2]
```diff
@@ -3,20 +3,205 @@ const expect = require('chai').expect;
...
+          min: 10,
+          max: 20,
+        },
+      });
+
+      const errors = validator.validate({});
+      expect(errors).to.have.length(0);
+    });
+
+    it('ignores the field if it does not have a rule decribed', () => {
```
Comment (SergeiKalachev) [inline 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js:68]: Not sure if it is the desired implementation. Should we fail if object comes with the field we have no rules for?

**Comment by jsru** [comment]
Решение было обновлено, посмотрим что скажет @tadjik1

File: 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js (hunk 1 above)
Comment (tadjik1) [reply 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js:50]: nope, this is unnecessary. the main idea is to get ourselves familiar with testing techniques :)

Comment (SergeiKalachev) [reply 04-testing-configuration-logging/01-unit-tests/test/Validator.test.js:50]: Ok, then this test doesn't hurt :)

File: 05-koajs/01-chat-app/app.js [hunk 3]
```diff
@@ -7,11 +7,43 @@ app.use(require('koa-bodyparser')());
...
 const Router = require('koa-router');
 const router = new Router();
+let incomingMessageResolve;
+let incomingMessagePromise = new Promise((resolve) => {
+  incomingMessageResolve = resolve;
+});
 
 router.get('/subscribe', async (ctx, next) => {
+  try {
+    const message = await incomingMessagePromise;
```
Comment (tadjik1) [inline 05-koajs/01-chat-app/app.js:17]: nicely done 👍

**Review by tadjik1 — APPROVED** [review]
thanks @SergeiKalachev , outstanding work 👍

**Pull request merged**